### PR TITLE
samples directory now passes the missing-function-docstring linter

### PR DESCRIPTION
**Pull request by macohen**
### Description
The samples directory has docstrings on all functions. This is progress on https://github.com/opensearch-project/opensearch-py/issues/610. The missing-function-docstring linter has not been enabled in setup.cfg because work in other directories is still needed and I wanted to break up the work into smaller commits; merges.

### Issues Resolved
N/A - progress on https://github.com/opensearch-project/opensearch-py/issues/610

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: samples/hello/hello.py [hunk 1]
```diff
@@ -16,6 +16,11 @@
 
 
 def main() -> None:
+    """
+    an example showing how to create an synchronous connection to OpenSearch, create an index, index a document
+    and search to return the document
+    :return:
```
Comment (saimedhi) [inline samples/hello/hello.py:22]: Hello @macohen,

Thanks for your contribution! I've noticed that `:return: `appears to be empty at several places. Could you please fill it or remove it if not essential?

**Review by saimedhi — CHANGES_REQUESTED** [review]

File: samples/hello/hello.py (hunk 1 above)
Comment (macohen) [reply samples/hello/hello.py]: done! thanks for the review. :)

Comment (saimedhi) [reply samples/hello/hello.py]: Linter is failing. Could you please run `nox -rs format`

Comment (macohen) [reply samples/hello/hello.py]: yeah. I saw that after I pushed. fixing log_collection_sample.py

**Review by saimedhi — APPROVED** [review]

**Pull request merged**